chirps/asset/templates/asset/dashboard.html

```diff
@@ -98,12 +98,14 @@ <h5 class="my-auto ml-0 mr-auto">{{ asset.name }}</h5>
         btn.addEventListener('click', async function () {
             const assetId = btn.getAttribute('data-asset-id');
             const response = await fetch(`/asset/ping/${assetId}/`, { headers: { 'X-Requested-With': 'XMLHttpRequest' } });
+            btn.innerHTML = '<i class="fa-solid fa-spinner fa-spin"></i>'
```

Comment:
please put this before the request to `/asset/ping/...` so that the spinner is shown after the button is clicked and before the response is received/handled

Reply:
Done! I don't know why I put it after the request, thanks for noticing.